### ISSUE TITLE
Problem with importing large dump file

### ISSUE THREAD

**Issue by sorin-pop** [djavorszky/ddn#91]
Try to import JCR_ALBATROS.sql (almost 10G) from r2d2 tmp/sorin into a mysql 5.6 database.

The upload counts up to 50% two times then:

This site can’t be reached

The connection was reset.

Can you please try this from your PC as well, Dani?

**Comment by djavorszky** [comment]
Yup, something definitely goes wrong. The following is logged:

```
2017/04/20 10:00:11 http: panic serving 192.168.211.1:38270: runtime error: invalid memory address or nil pointer dereference
goroutine 115164 [running]:
net/http.(*conn).serve.func1(0xc4202c74a0)
	/usr/local/go/src/net/http/server.go:1721 +0xd0
panic(0x821c40, 0xaa2bb0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
main.importAction(0xa7eac0, 0xc42041a7e0, 0xc4205c4700)
	/home/javdaniel/go/src/github.com/djavorszky/ddn/server/handlers.go:62 +0x2d2
net/http.HandlerFunc.ServeHTTP(0x8bafe8, 0xa7eac0, 0xc42041a7e0, 0xc4205c4700)
	/usr/local/go/src/net/http/server.go:1942 +0x44
github.com/djavorszky/ddn/common/srv.Logger.func1(0xa7eac0, 0xc42041a7e0, 0xc4205c4700)
	/home/javdaniel/go/src/github.com/djavorszky/ddn/common/srv/srv.go:14 +0x9c
net/http.HandlerFunc.ServeHTTP(0xc42012f700, 0xa7eac0, 0xc42041a7e0, 0xc4205c4700)
	/usr/local/go/src/net/http/server.go:1942 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc42011a870, 0xa7eac0, 0xc42041a7e0, 0xc4205c4700)
	/home/javdaniel/go/src/github.com/gorilla/mux/mux.go:114 +0x10c
net/http.serverHandler.ServeHTTP(0xc42018c000, 0xa7eac0, 0xc42041a7e0, 0xc420327d00)
	/usr/local/go/src/net/http/server.go:2568 +0x92
net/http.(*conn).serve(0xc4202c74a0, 0xa7f240, 0xc4205178c0)
	/usr/local/go/src/net/http/server.go:1825 +0x612
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2668 +0x2ce
```

I'm not entirely sure why though - the panic comes from [here](https://github.com/djavorszky/ddn/blob/master/server/handlers.go#L62), but I'm not sure why - it works with smaller files.

I just tested it locally, after about a minute of upload, it is uploaded successfully, so I believe the root cause of the issue is that the upload takes too long and the web server in front drops the connection.

I'll look into what we could do - e.g. see if we can upload in an async way or something along that line - I'll also talk with the sysadmin about it.

**Comment by sorin-pop** [comment]
Hi Dani,

Any news on this?

**Comment by djavorszky** [comment]
We're working on it. With an increased timeout, the upload got to 99%, then failed. Our sysadmin expressed concerns about modifying the webserver of that particular server, as it has some public facing sites as well, and he doesn't want to take chances.

As a next step, we're moving the ddn server to another server instance over which we have more authority and control, should be done later today or early tomorrow.

Note though, that the configuration files of the connectors may need to be updated to point to the new address. I'll let you know once there's more progress.

**Comment by djavorszky** [comment]
I've tested this further:
Works locally without webserver
Works locally with webserver
Does not work remote without webserver
Does not work remote without webserver

It looks like at the 30 second mark the connection is reset - most probably network configuration.

**Comment by sorin-pop** [comment]
Hi Zsolt,

Could you help us in getting Péter Erdei to have a look at this? It seems
to be a problem caused by our internal network settings, so maybe he could
help.

https://github.com/djavorszky/ddn/issues/91

With this problem the CloudDB will be pretty much useless when we want to
import larger dumps, which can frequently be the case.

Regards.

**Comment by djavorszky** [comment]
Alternatively, I'm thinking about implementing #66 or some kind of async upload, if that could work.

**Comment by djavorszky** [comment]
Feature is implemented (see #66) and live
